src/quantum/azext_quantum/_client_factory.py

```diff
@@ -0,0 +1,51 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+
+import os
+
+
+def is_env(name):
+    return 'AZURE_QUANTUM_ENV' in os.environ and os.environ['AZURE_QUANTUM_ENV'] == name
+
+
+def base_url():
+    if 'AZURE_QUANTUM_BASEURL' in os.environ:
+        return os.environ['AZURE_QUANTUM_BASEURL']
+    if is_env('canary'):
+        return "https://app-jobs-canarysouthcentralus.azurewebsites.net/"
+    return "https://app-jobscheduler-prod.azurewebsites.net/"
+
+
+def _get_data_credentials(cli_ctx, subscription_id=None):
+    from azure.cli.core._profile import Profile
+    profile = Profile(cli_ctx=cli_ctx)
+    creds, _, _ = profile.get_login_credentials(subscription_id=subscription_id, resource="https://quantum.microsoft.com")
```

Comment:
again, will this login in be different in different cloud?

Reply:
We're just working on entering private-preview and have no short term plans to support anything but public cloud.